client.go

```diff
@@ -336,7 +340,19 @@ func (c *Client) send(call *Call) {
 	// Encode and send the request.
 	c.request.Seq = seq
 	c.request.Method = call.Method
-	err := c.codec.WriteRequest(&c.request, call.Args)
+	stop := make(chan struct{})
+	callArgs := call.Args
+	var err error
+	go func() {
+		defer close(stop)
+		err = c.codec.WriteRequest(&c.request, callArgs)
```

Comment:
No concerns about about the concurrent access to `err` or possible leakage of this inline function? Have you tried `SetDeadline\SetWriteDeadline`?
```
	// SetDeadline sets the read and write deadlines associated
	// with the connection. It is equivalent to calling both
	// SetReadDeadline and SetWriteDeadline.
	//
	// A deadline is an absolute time after which I/O operations
	// fail instead of blocking. The deadline applies to all future
	// and pending I/O, not just the immediately following call to
	// Read or Write. After a deadline has been exceeded, the
	// connection can be refreshed by setting a deadline in the future.
	//
	// If the deadline is exceeded a call to Read or Write or to other
	// I/O methods will return an error that wraps os.ErrDeadlineExceeded.
	// This can be tested using errors.Is(err, os.ErrDeadlineExceeded).
	// The error's Timeout method will return true, but note that there
	// are other possible errors for which the Timeout method will
	// return true even if the deadline has not been exceeded.
	//
	// An idle timeout can be implemented by repeatedly extending
	// the deadline after successful Read or Write calls.
	//
	// A zero value for t means I/O operations will not time out.
	SetDeadline(t time.Time) error
...
	// SetWriteDeadline sets the deadline for future Write calls
	// and any currently-blocked Write call.
	// Even if write times out, it may return n > 0, indicating that
	// some of the data was successfully written.
	// A zero value for t means Write will not time out.
	SetWriteDeadline(t time.Time) error
	```